src/background/api.ts

```diff
@@ -1,93 +1,78 @@
-function tweetFormData(target: FeedbackTarget): FormData {
-  const tweetData = new FormData()
+import { JsonDecoder } from 'ts.data.json'
 
-  const status = target.feedbackState.editorState.getCurrentContent().getPlainText('\u0001')
-  tweetData.append('status', status)
-
-  if (target.feedbackTargetType === 'tab') {
-    tweetData.append('domain', target.domain!)
-  } else {
-    tweetData.append('help', 'true')
-  }
-
-  // Adding all the screenshot files under the same form key - 'images'.
-  target.feedbackState.images.forEach(image => tweetData.append('images', image.blob, image.name))
-
-  return tweetData
-}
-
-function makeTweetRequest(formData: FormData): Promise<Response> {
-  return fetch(`${window.roarServerUrl}/v1/feedback`, {
-    method: 'POST',
-    credentials: 'include',
-    body: formData,
-  })
-}
-
-export const postTweet = async (target: FeedbackTarget, chrome: typeof global.chrome, dispatchBackgroundActions: Dispatchers<BackgroundAction>) => {
-  const targetId: FeedbackTargetId = target.id
+export type FetchRoarResult<T> =
+  | { ok: true; data: T }
+  | { ok: false; reason: 'response not json'; text: string; details: string }
+  | { ok: false; reason: 'response not expected data type'; text: string; details: string }
+  | { ok: false; reason: 'bad request'; details: string }
+  | { ok: false; reason: 'unauthorized'; details: string }
+  | { ok: false; reason: 'server down'; details: string }
+  | { ok: false; reason: 'unknown status'; details: string; status: number }
+  | { ok: false; reason: 'network down'; details: string }
 
+export async function fetchRoar<T extends object>(path: string, init: RequestInit, decoder: JsonDecoder.Decoder<T>): Promise<FetchRoarResult<T>> {
   try {
-    dispatchBackgroundActions.postTweetStart({ targetId })
-    const res = await makeTweetRequest(tweetFormData(target))
-    if (res.status !== 201) {
-      return dispatchBackgroundActions.postTweetFailure({
-        targetId,
-        error: { message: await res.text() },
-      })
+    const response = await fetch(`${window.roarServerUrl}/${path}`, { credentials: 'include', ...init })
```

Comment:
We should probably use axios or [add timeout functionality](https://stackoverflow.com/a/57888548/11899960)